keras_cv/models/segmentation/deeplab.py

```diff
@@ -64,89 +76,89 @@ def __init__(
+                    include_top=False,
+                    name="resnet50v2",
+                    weights=parse_weights(weights, False, "resnet50v2"),
+                    pooling=None,
+                    **kwargs,
                 )
-                backbone = backbone.as_backbone()
-                self.backbone = backbone
+
         else:
             # TODO(scottzhu): Might need to do more assertion about the model
             if not isinstance(backbone, tf.keras.layers.Layer):
                 raise ValueError(
                     "Backbone need to be a `tf.keras.layers.Layer`, "
                     f"received {backbone}"
                 )
-            self.backbone = backbone
+        self.backbone = backbone
 
-        # ================== decoder ==================
-        if isinstance(decoder, str):
-            # TODO(scottzhu): Add ASPP decoder.
-            supported_premade_decoder = ["fpn"]
-            if decoder not in supported_premade_decoder:
-                raise ValueError(
-                    "Supported premade decoder are: "
-                    f'{supported_premade_decoder}, received "{decoder}"'
-                )
-            self._decoder_passed = decoder
-            if decoder == "fpn":
-                # Infer the FPN level from the backbone. If user need to customize
-                # this setting, they should manually create the FPN and backbone.
-                if not isinstance(backbone.output, dict):
-                    raise ValueError(
-                        "Expect the backbone's output to be dict, "
-                        f"received {backbone.output}"
-                    )
-                backbone_levels = list(backbone.output.keys())
-                min_level = backbone_levels[0]
-                max_level = backbone_levels[-1]
-                decoder = keras_cv.layers.FeaturePyramid(
-                    min_level=min_level, max_level=max_level
-                )
-
-        # TODO(scottzhu): do more validation for the decoder when we have a common
-        # interface.
-        self.decoder = decoder
+        if spatial_pyramid_pooling is None:
+            self.aspp = SpatialPyramidPooling(dilation_rates=[6, 12, 18])
+        else:
+            self.aspp = spatial_pyramid_pooling
 
         self._segmentation_head_passed = segmentation_head
         if segmentation_head is None:
-            # Scale up the output when using FPN, to keep the output shape same as the
-            # input shape.
-            if isinstance(self.decoder, keras_cv.layers.FeaturePyramid):
-                output_scale_factor = pow(2, self.decoder.min_level)
-            else:
-                output_scale_factor = None
-
-            segmentation_head = (
-                keras_cv.models.segmentation.__internal__.SegmentationHead(
-                    classes=classes, output_scale_factor=output_scale_factor
-                )
+            segmentation_head = tf.keras.Sequential(
+                [
```

Comment:
I vaguely remember there are some conv2d + BN needed before the final classification head. Did u find that in the original papar/code?

Reply:
From the paper:
"The resulting features from all the branches are then concatenated and pass through another 1x1 conv (also with 256 filters and batch norm) before the final 1x1 conv which generates the final logits"

From [Gluon](https://cv.gluon.ai/_modules/gluoncv/model_zoo/deeplabv3.html), it's a single 3x3 conv

From previous segmentation head, it's 2 3x3 conv layers.

They all seem very different. And TBH I don't see how this adds any value given it's mapping from 256 channels to 256 channels. But I chose to stick with what the paper says regardless.

Reply:
Some update -- adding another conv_bn_relu makes the meanIOU drop by 1.0. But sticking to the paper sounds fine with me.